client/podman/image.go

```diff
@@ -81,9 +85,79 @@ func (p *PodmanRestClient) ImagePull(id string) error {
 		Context:            params.Context,
 		Client:             params.HTTPClient,
 	}
-	_, err := p.RestClient.Submit(op)
+	res, err := p.RestClient.Submit(op)
 	if err != nil {
 		return fmt.Errorf("error pulling image %s: %v", id, err)
 	}
+	// eventually err is nil but auth problems are reported as json after a string msg
+	if resStr, ok := res.(string); ok {
+		resStrClean := strings.TrimLeftFunc(resStr, func(r rune) bool {
+			if r == '{' {
+				return false
+			}
+			return true
+		})
+		var jsonRes map[string]interface{}
+		if err = json.Unmarshal([]byte(resStrClean), &jsonRes); err == nil {
+			if errMsg, ok := jsonRes["error"]; ok && errMsg != "" {
+				return fmt.Errorf("unable to pull image %s: %v", id, errMsg)
+			}
+		}
+	}
+	return nil
+}
+
+func GetXRegistryAuth(image string) *string {
```

Comment:
Is this intended to be an exported method? (Or could it be getXRegistryAuth() if only use is within this package?)